src/main/java/org/folio/spring/service/TenantService.java

```diff
@@ -0,0 +1,58 @@
+package org.folio.spring.service;
+
+import static org.apache.commons.lang3.BooleanUtils.isTrue;
+
+import java.sql.ResultSet;
+import liquibase.exception.LiquibaseException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.spring.FolioExecutionContext;
+import org.folio.spring.exception.NotFoundException;
+import org.folio.spring.liquibase.FolioSpringLiquibase;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Service;
+
+@Log4j2
+@RequiredArgsConstructor
+@Service
+public class TenantService {
+  private static final String DESTROY_SQL = "DROP SCHEMA IF EXISTS %1$s CASCADE; DROP ROLE IF EXISTS %1$s";
+  private static final String EXIST_SQL = "SELECT EXISTS(SELECT 1 FROM pg_namespace WHERE nspname=?)";
+
+  private final JdbcTemplate jdbcTemplate;
+  private final FolioExecutionContext context;
+  private final FolioSpringLiquibase folioSpringLiquibase;
+
+  public void createTenant(String tenant) throws LiquibaseException {
+    if (folioSpringLiquibase != null) {
+      var tenantId = context.getTenantId();
+      var schemaName = context.getFolioModuleMetadata().getDBSchemaName(tenantId);
+
+      folioSpringLiquibase.setDefaultSchema(schemaName);
+      log.info("About to start liquibase update for tenant [{}]", tenant);
+      folioSpringLiquibase.performLiquibaseUpdate();
+      log.info("Liquibase update for tenant [{}] executed successfully", tenant);
+    }
+  }
+
+  /**
+   * @throws NotFoundException when tenant not found.
+   */
+  public void deleteTenant(String tenant) {
```

Comment:
Can we get the tenant name from `FolioExecutionContext` there and below?

Reply:
What is the purpose of such change? Is it possible to have different tenant id in context vs in headers? 

IMHO passing the tenant directly makes code clearer.

Reply:
Here we are forced to follow the API required by OKAPI. That is why we have that ambiguity here when TenantId is passed through parameters and also is available in the FolioContext. In this case, it does not matter which one to use. I would recommend just checking that those two are equal.
But in all other APIs, we should not pass TenantId through parameters in methods. It must be retrieved only from FolioContext.

Reply:
Thanks @psmagin and @TarasSpashchenko, I have updated code to use tenant from context.